### PR TITLE
stop periodic resync on endpoints object

### PR DESCRIPTION
**Pull request by freehan**

### CODE REVIEW

File: pkg/context/context.go [hunk 1]
```diff
@@ -144,12 +144,15 @@ func NewControllerContext(
 		IngressInformer:         informernetworking.NewIngressInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
 		ServiceInformer:         informerv1.NewServiceInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
 		BackendConfigInformer:   informerbackendconfig.NewBackendConfigInformer(backendConfigClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		EndpointInformer:        informerv1.NewEndpointsInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		PodInformer:             informerv1.NewPodInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		NodeInformer:            informerv1.NewNodeInformer(kubeClient, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		SvcNegInformer:          informersvcneg.NewServiceNetworkEndpointGroupInformer(svcnegClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		recorders:               map[string]record.EventRecorder{},
-		healthChecks:            make(map[string]func() error),
+		// Do not trigger periodic resync on Endpoints object,
```
Comment (prameshj) [inline pkg/context/context.go:147]: nit: replace ending "," with "."

File: pkg/context/context.go [hunk 2]
```diff
@@ -144,12 +144,15 @@ func NewControllerContext(
...
+		// This aims improve NEG controller performance by avoiding unnecessary NEG sync that triggers for each NEG syncer.
+		// As periodic resync may temporary starve NEG API ratelimit quota.
+		EndpointInformer: informerv1.NewEndpointsInformer(kubeClient, config.Namespace, 0, utils.NewNamespaceIndexer()),
+		PodInformer:      informerv1.NewPodInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
```
Comment (prameshj) [inline pkg/context/context.go:151]: Do we need resync for Pods?

Comment (swetharepakula) [reply pkg/context/context.go:151]: The only thing I see that pod triggers is the readiness reflector. Would there be any issues if that doesn't periodically resync?

Comment (freehan) [reply pkg/context/context.go:151]: Good question. 

Readiness reflector relies on pod periodic resyncs to mark Pod to ready when timeout waiting for pods to become healthy behind GCLB.

Comment (freehan) [reply pkg/context/context.go:151]: I am thinking if we should disable resync for nodes as well.

File: pkg/context/context.go [hunk 3]
```diff
@@ -144,12 +144,15 @@ func NewControllerContext(
...
-		NodeInformer:            informerv1.NewNodeInformer(kubeClient, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		SvcNegInformer:          informersvcneg.NewServiceNetworkEndpointGroupInformer(svcnegClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
-		recorders:               map[string]record.EventRecorder{},
-		healthChecks:            make(map[string]func() error),
+		// Do not trigger periodic resync on Endpoints object.
+		// This aims improve NEG controller performance by avoiding unnecessary NEG sync that triggers for each NEG syncer.
+		// As periodic resync may temporary starve NEG API ratelimit quota.
+		EndpointInformer: informerv1.NewEndpointsInformer(kubeClient, config.Namespace, 0, utils.NewNamespaceIndexer()),
+		PodInformer:      informerv1.NewPodInformer(kubeClient, config.Namespace, config.ResyncPeriod, utils.NewNamespaceIndexer()),
+		NodeInformer:     informerv1.NewNodeInformer(kubeClient, config.ResyncPeriod, utils.NewNamespaceIndexer()),
```
Comment (prameshj) [inline pkg/context/context.go:152]: Can we do the same for nodes too?

Comment (prameshj) [reply pkg/context/context.go:152]: Doing this wont change much w.r.t performance since NEG controller only processes Node adds and deletes.

**Pull request merged**